pkg/operator/ceph/cluster/crash/reconcile.go

```diff
@@ -175,6 +177,12 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 			}
 		}
 
+		secret := &corev1.Secret{}
```

Comment:
This check should be moved above to line 160, right after the `cephCluster.Spec.CrashCollector.Disable` condition.